File: React/models/SessionInfo.ts
import User from './User';

export type ModulePermission = {
    name: string;
    canView: boolean;
    canEdit: boolean;
};

class SessionInfo {
    systemDate: Date;
    user: User;
    permissions: ModulePermission[];

    constructor(systemDate: Date, user: User, permissions: ModulePermission[]) {
        this.systemDate = systemDate;
        this.user = user;
        this.permissions = permissions;
    }
}

export default SessionInfo;

File: Java/actinver-fw-interno/FiduciaWeb/src/mx/com/inscitech/fiducia/dtos/SessionInfo.java
package mx.com.inscitech.fiducia.dtos;

import java.util.List;

public class SessionInfo {
    
    private SessionUser user;
    private String systemDate;
    private List<SessionUserPermisions> permissions;
    
    public SessionInfo() {
        super();
    }

    public SessionInfo(SessionUser user, String systemDate, List<SessionUserPermisions> permissions) {
        this.user = user;
        this.systemDate = systemDate;
        this.permissions = permissions;
    }

    public void setUser(SessionUser user) {
        this.user = user;
    }

    public SessionUser getUser() {
        return user;
    }

    public void setSystemDate(String systemDate) {
        this.systemDate = systemDate;
    }

    public String getSystemDate() {
        return systemDate;
    }

    public void setPermissions(List<SessionUserPermisions> permissions) {
        this.permissions = permissions;
    }

    public List<SessionUserPermisions> getPermissions() {
        return permissions;
    }
}
